test/BEST_PRACTICES.md

```diff
@@ -0,0 +1,54 @@
+# Test Best Practices
+
+This is a collection of best practices to write high quality tests. The best practices
+can be referenced in review comments to speed up the review process.
+
+- [Go Style](#go-style)
+- [Test Case Traceability](#test-case-traceability)
+- [Don't Sleep](#dont-sleep)
+- [Independent](#independent)
+- [Secrets](#secrets)
+- [Logging](#logging)
+
+## Go Style
+
+We don't want to redefine best practices for go, therefore we should reuse the existing ones
```

Comment:
The reason here is that we want these best practices to focus on test design and implementation. The same coding standards used throughout the product team should also be followed for test code.

Reply:
The problem is that the project is completely missing a Contribution or Best Practices guide, so I don't know which shared practices eng are using

Reply:
@ldimaggi tried to reword it wdyt?

Reply:
```suggestion
Test code style to be consistent with that of integreatly-operator codebase [TODO: link to golang practices used by engineering]
```
perhaps something like the above?
tbh if there are no shared practices to link to, I would question the value of this section in the doc though.

Reply:
Try to reworded again, wdty?

Reply:
ping @damienomurchu

Reply:
Looks good. Were you able to locate the Eng Golang practices doc?

Reply:
I wasn't when I looked, but lets not let this block this PR. 
@b1zzu make the above change with the TODO, and we can update this doc later after its merged?